Lib/ufo2ft/outlineOTF.py

```diff
@@ -584,6 +605,101 @@ def setupTable_hhea(self):
         # glyph count
         hhea.numberOfHMetrics = len(self.allGlyphs)
 
+    def setupTable_vmtx(self):
+        """
+        Make the vmtx table.
+
+        **This should not be called externally.** Subclasses
+        may override or supplement this method to handle the
+        table creation in a different way if desired.
+        """
+
+        self.otf["vmtx"] = vmtx = newTable("vmtx")
+        vmtx.metrics = {}
+        for glyphName, glyph in self.allGlyphs.items():
+            height = glyph.height
+            verticalOrigin = _getVerticalOrigin(glyph)
+            top = 0
+            if len(glyph) or len(glyph.components):
+                # tsb should be consistent with glyf yMax, which is just
+                # maximum y for coordinate data
+                pen = ControlBoundsPen(self.ufo, ignoreSinglePoints=True)
+                glyph.draw(pen)
+                if pen.bounds is not None:
+                    top = pen.bounds[3]
+            # take ceil of tsb/yMax, as fontTools does with max bounds
+            vmtx[glyphName] = (_roundInt(height),
+                               int(math.ceil(verticalOrigin) - math.ceil(top)))
+
+    def setupTable_VORG(self):
+        """
+        Make the VORG table.
+
+        **This should not be called externally.** Subclasses
+        may override or supplement this method to handle the
+        table creation in a different way if desired.
+        """
+        self.otf["VORG"] = vorg = newTable("VORG")
+        vorg.majorVersion = 1
+        vorg.minorVersion = 0
+        vorg.VOriginRecords = {}
+        # Find the most frequent verticalOrigin
+        vorg_count = Counter(_getVerticalOrigin(glyph)
+                             for glyph in self.allGlyphs.values())
+        vorg.defaultVertOriginY = vorg_count.most_common(1)[0][0]
+        if len(vorg_count) > 1:
+            for glyphName, glyph in self.allGlyphs.items():
+                vorg.VOriginRecords[glyphName] = glyph.verticalOrigin
+        vorg.numVertOriginYMetrics = len(vorg.VOriginRecords)
+
+    def setupTable_vhea(self):
+        """
+        Make the vhea table.
+
+        **This should not be called externally.** Subclasses
+        may override or supplement this method to handle the
+        table creation in a different way if desired.
+        """
+        self.otf["vhea"] = vhea = newTable("vhea")
+        font = self.ufo
+        head = self.otf["head"]
+        vhea.tableVersion = 1.1
```

Comment:
It should be `0x00011000` for v1.1, instead of `1.1`. see also https://github.com/behdad/fonttools/issues/540